### ISSUE TITLE
Flow syntax callToFunc<SomeType>(arg1, arg2) is unsupported (function _call_, not definition)

### ISSUE THREAD

**Issue by lll000111** [gajus/eslint-plugin-flowtype#356]
In Flow it is possible to specify a generic type for a function that uses a generic  type definition when actually _calling_ the function. Often that is unnecessary and/or useless, but occasionally, when Flow can't decide on its own, it helps to tell Flow explicitly what the generic part is supposed to be (it is still checked by Flow if it makes sense, of course).

This syntax does not seem to be supported, I get

    error  Infix operators must be spaced  space-infix-ops

when I write something like

```js
const obj = await StorageVersionedObjects.getObjectByIdObj<ModuleObj>({....})
```

**Comment by pnevyk** [comment]
It seems that Babel parser doesn't support this syntax, but it probably should since Flow itself parses it correctly. Looking at the Babel AST, the code `foo<Foo>({})` parses into `identifier binary_op identifier binary_op object_expr` (hence the "infix operators" lint error), and the code `foo<Foo>()` causes syntax error. Please, discuss this issue in [Babel](https://github.com/babel/babel/tree/master/packages/babel-parser) repository itself (you should perhaps search existing issues first).

**Comment by lll000111** [comment]
> Flow is not loaded by anything

I did not say anything in that direction?

> Nevertheless, this issue is not related to eslint-plugin-flowtype. T

My question was, see my reply, what Babel is actually being used? By whom? And which Flow plugin?

If Babel 7 works it should be eslint's fault for using an old Babel?

**Comment by lll000111** [comment]
Okay... so I look at [eslint-plugin-flowtype/package.json](https://github.com/gajus/eslint-plugin-flowtype/blob/master/package.json#L14):

It has

    "babel-eslint": "^6.1.2",

The current version [of babel-eslint is 9.0.0](https://github.com/babel/babel-eslint/releases). The Babel version is in [babel-eslint/package.json](https://github.com/babel/babel-eslint/blob/master/package.json#L15) and it is 7.0 - in the current version.

My current interpretation is that this plugin here uses an old version of eslint-babel which uses an old version of Babel, and that upgrading the dependency would solve the problem I'm reporting, since current Babel understands the syntax, if I interpret that correctly?

**Comment by lll000111** [comment]
As an aside, [flow-remove-types](https://github.com/flowtype/flow-remove-types) doesn't understand this syntax either and leaves the type in the code :-(

**Comment by pnevyk** [comment]
It uses quite old version of [babel parser](https://github.com/flowtype/flow-remove-types/blob/master/package.json#L42) (it was a separate package called babylon before) and this was not supported at that time probably.

**Comment by pnevyk** [comment]
@rostislav-simonik And what version of babel-eslint do you use? Because an older version of babel-eslint might use an older version of babel, ie. less than 7. And what is the source code where eslint gives you this parsing error?

**Comment by rostislav-simonik** [comment]
@pnevyk oh you are right, upgrade of everything (babel-eslint, eslint, eslint-plugins) to recent version helped.

**Comment by FermiDirak** [comment]
any update on this?

**Comment by pnevyk** [comment]
@FermiDirak Imho there cannot be an update on this, this error is coming from the parser, not from this plugin. The solution is to upgrade babel parser (babel-eslint package, maybe others).

@lll000111 Is this issue resolved for you?

**Comment by lll000111** [comment]
Yes, solved.

**Comment by FermiDirak** [comment]
Solved for me too. The solution is to as you said upgrade babel parser.
